packages/next/build/webpack/plugins/build-manifest-plugin.ts

```diff
@@ -84,14 +91,42 @@ export default class BuildManifestPlugin {
           assetMap.pages['/'] = assetMap.pages['/index']
         }
 
+        // Add the runtime buildManifest.js file (generated later in this file) as a
+        // dependency for the app. If the flag is false, the file won't be downloaded
+        // by the client
+        if (this.options.clientManifest) {
+          assetMap.pages['/_app'].push(
+            `${CLIENT_STATIC_FILES_PATH}/${
+              this.options.buildId
+            }/_buildManifest.js`
+          )
+        }
+
         assetMap.pages = Object.keys(assetMap.pages)
           .sort()
           // eslint-disable-next-line
           .reduce((a, c) => ((a[c] = assetMap.pages[c]), a), {} as any)
 
+        const manifestContentString = JSON.stringify(assetMap, null, 2)
```

Comment:
Might be worth using [devalue](https://www.npmjs.com/package/devalue) here instead of JSON.stringify, it'll produce more compact output.

Reply:
Took a quick look at the npm docs--what does devalue do to compress the content? I'm already removing whitespace, and I don't think there's an XSS risk here.

Reply:
Mainly drops quoted keys. Agreed though, without the risk of XSS it's not a huge win.

Reply:
In that case I'll close this for the time being, and make a note of it in the doc for possible future optimizations